proposals/0165-async-vote-execution.md

```diff
@@ -0,0 +1,120 @@
+
+With different hardware and running environment, there will always be some
+difference on speed of transaction execution between validators. Generally
+speaking, because vote transactions are so simple, the variation between vote
+execution should be smaller than that between non-vote executions. Therefore,
+if we only execute vote transactions in a block before voting on the block,
+it is more likely validators can reach consensus faster.
+
+Even with async vote execution, forks can still happen because of
+various other situations, like network partitions or mis-configured validators.
+This work just reduces the chances of forks caused by variance in non-vote
+transaction executions.
+
+The non-vote transactions do need to be executed eventually. Even though it's
+hard to make sure everyone executes every block within 400ms, on average majority
+of the cluster should be able to keep up.
+
+## New Terminology
+
+- `VED`: Vote Execution Domain, Vote transactions and all its dependencies (e.g.
+fee payers for votes).
+- `VED Bankhash`: The hash calculated after executing only vote transactions in
+a block. If there are no votes, use default hash.
+- `UED`: User Execution Domain, currently everything other than votes. We may
+have more domains in the future.
+- `UED Bankhash`: The hash calculated after executing only non-vote transactions
+in a block. If there are no non-vote transactions, use default hash.
+
+## Detailed Design
+
+### Allow leader to skip execution of transactions (Bankless Leader)
+
+There is already on-going effort to totally skip execution of all transactions
+when leader pack new blocks. See SIMD 82, SIMD 83, and related trackers:
+https://github.com/anza-xyz/agave/issues/2502
+
+Theoretically we could reap some benefit without Bankless Leader, the leader
+pack as normal, while other validators only replay votes first, then later
+execute other transactions and compare with the bankhash of the leader. But in
+such a setup we gain smaller speedup without much benefits, it is a possible
+route during rollouts though.
+
+### Separating vote transactions and dependencies into a different domain
+
+To make sure vote transactions can be executed independently, we need to
+isolate its dependencies.
+
+#### Remove clock program's dependency on votes
+
+Introduce new transaction `ClockBump` to remove current clock program's
+dependency on votes.
+
+The transaction `ClockBump` is sent by a leader with at least 0.5% stake
+every 12 slots to correct the clock drift. A small script can be used to
+refund well-behaving leaders the cost of the transactions.
+
+#### Split vote accounts into two accounts in VED and UED respectively
+
+We need to allow users move money in and out of the vote accounts, but
+we also need the vote accounts to vote in VED. So there will be two accounts:
+
+- `VoteTowerAccount`: tracks tower state and vote authority, it will be
+in `VED`, it is updated by vote transactions and tracks vote credits.
```

Comment:
A flow diagram/description of how users will pay for vote fees would be useful here. A step by  step from vote account creation, to funding the account, to which domains it's created/moved into, etc.